### PR TITLE
Allow setting number of neighbor cells to walk

### PR DESCRIPTION
**Pull request by brettstack**
This is breaking, since it modifies the function signature of Heartbeat, but I think it's best for callback to be at the end.

### CODE REVIEW

File: poke.io.js [hunk 1]
```diff
@@ -239,17 +239,17 @@ function Pokeio() {
     };
 
     // IN DEVELPOMENT, YES WE KNOW IS NOT WORKING ATM
-    self.Heartbeat = function (callback) {
+    self.Heartbeat = function (numNeighborCells, callback) {
+        numNeighborCells = numNeighborCells === undefined ? 10 : numNeighborCells
```
Comment (servel333) [inline poke.io.js]: `numNeighborCells` will be a function (callback) if it is not specified.

This code would require calling as `Heartbeat(undefined, function(...){...})`

Recommend checking if it's a function:

```
if (typeof(numNeighborCells) === "function") {
  callback = numNeighborCells;
  numNeighborCells = 10;
}
```

Comment (brettstack) [inline poke.io.js]: Good catch. I will update when I get to the office.

**Comment by Elyx0** [comment]
Doesn't the heartbeat walk expect exactly 21cells ? I'm a bit concerned that custom walk might get easy to flag in Niantic side for bans.

**Comment by brettstack** [comment]
Updated so default is 21

**Comment by jdpnielsen** [comment]
Whats the use case for custom walk length, if as @Elyx0 says the heartbeat expects 21 cells?

**Comment by brettstack** [comment]
If you are only interested in the 1 cell, I expect the execution time on
the pogo server would be less if you only request 1 cell instead of 21. I
did not validate this assumption. Response size is at least smaller, which
does impact speed.

**Pull request closed without merging**